ambari-server/src/main/resources/Ambari-DDL-MySQL-CREATE.sql

```diff
@@ -635,12 +649,17 @@ CREATE TABLE hostgroup (
   CONSTRAINT FK_hg_blueprint_name FOREIGN KEY (blueprint_name) REFERENCES blueprint(blueprint_name));
 
 CREATE TABLE hostgroup_component (
-  blueprint_name VARCHAR(100) NOT NULL,
-  hostgroup_name VARCHAR(100) NOT NULL,
-  name VARCHAR(100) NOT NULL,
-  provision_action VARCHAR(100),
-  CONSTRAINT PK_hostgroup_component PRIMARY KEY (blueprint_name, hostgroup_name, name),
-  CONSTRAINT FK_hgc_blueprint_name FOREIGN KEY (blueprint_name, hostgroup_name) REFERENCES hostgroup(blueprint_name, name));
+  id BIGINT NOT NULL,
+  blueprint_name VARCHAR(255) NOT NULL,
+  hostgroup_name VARCHAR(255) NOT NULL,
+  name VARCHAR(255) NOT NULL,
+  mpack_name VARCHAR(255),
+  mpack_version VARCHAR(100),
+  service_name VARCHAR(255),
+  provision_action VARCHAR(255),
+  CONSTRAINT PK_hostgroup_component PRIMARY KEY (id),
+  CONSTRAINT FK_hgc_blueprint_name FOREIGN KEY (blueprint_name, hostgroup_name) REFERENCES hostgroup (blueprint_name, name),
+  CONSTRAINT FK_hgc_blueprint_service FOREIGN KEY (blueprint_service_id) REFERENCES blueprint_service(id));
```

Comment:
```
Key column 'blueprint_service_id' doesn't exist in table
```

Reply:
Good catch. This FK is no more needed, will drop.

Reply:
Two more of these still left in the Derby and SQLServer scripts.

Reply:
Done.